### ISSUE TITLE
High cpu while parsing http request with multipart form data

### ISSUE THREAD

**Issue by rajaskunte** [playframework/playframework#11693]
### Play Version

Play - 2.5.10

### API

NA

### Operating System

Linux 5.10.0-0.deb10.16-cloud-amd64 SMP Debian 5.10.127-2~bpo10+1 (2022-07-28) x86_64 GNU/Linux

### JDK

openjdk version "1.8.0_342"
OpenJDK Runtime Environment (build 1.8.0_342-b07)
OpenJDK 64-Bit Server VM (build 25.342-b07, mixed mode)

### Expected Behavior

Http request with multipart body should not result in cpu spike. 

### Actual Behavior

We have exposed a http public endpoint which accepts email content as multipart form data. 
We see at time while processing such requests cpu utilization shoots to 100%. This happens for random requests and not for all. Below is the stack trace; we see the dispatcher thread is stuck in runnable state for any time varying from 2 mins to 70 mins. Also we do not see any our application code invoked. 

Stack Trace - 

```
"application-akka.actor.default-dispatcher-226" #715 prio=5 os_prio=0 tid=0x00007ff744021800 nid=0x302 runnable [0x00007ff7213e0000]
   java.lang.Thread.State: RUNNABLE
	at akka.util.ByteString$ByteStrings.apply(ByteString.scala:411)
	at play.core.parsers.Multipart$.play$core$parsers$Multipart$$byteAt(Multipart.scala:223)
	at play.core.parsers.Multipart$BoyerMoore.rec$3(Multipart.scala:542)
	at play.core.parsers.Multipart$BoyerMoore.nextIndex(Multipart.scala:548)
	at play.core.parsers.Multipart$BodyPartParser.handleDataPart(Multipart.scala:388)
	at play.core.parsers.Multipart$BodyPartParser$$anonfun$handleDataPart$1.apply(Multipart.scala:407)
	at play.core.parsers.Multipart$BodyPartParser$$anonfun$handleDataPart$1.apply(Multipart.scala:407)
	at play.core.parsers.Multipart$BodyPartParser$$anonfun$continue$1.apply(Multipart.scala:447)
	at play.core.parsers.Multipart$BodyPartParser$$anonfun$continue$1.apply(Multipart.scala:447)
	at play.core.parsers.Multipart$BodyPartParser.onPush(Multipart.scala:267)
	at play.core.parsers.Multipart$BodyPartParser.onPush(Multipart.scala:238)
	at akka.stream.stage.AbstractStage$PushPullGraphLogic$$anon$1.onPush(Stage.scala:55)
	at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:747)
	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:649)
	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:471)
	at akka.stream.impl.fusing.GraphInterpreterShell.receive(ActorGraphInterpreter.scala:410)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:603)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:618)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:484)
	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:529)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
	at akka.actor.ActorCell.invoke(ActorCell.scala:495)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

### Reproducible Test Case

Unable to reproduce as we don't have the request data.

**Comment by mkurz** [comment]
I see you still use Play 2.5.10 which is EOL since long time already. Please upgrade to latest Play 2.8.
There have been fixes to the multipart form data recently and I would guess that probably would fix your problem too.

**Comment by rajaskunte** [comment]
Thanks. The issue was resolved after upgrading to Play v2.8.13.
